### PR TITLE
r2-cutter: init at 1.1(.0), radare2 qt gui

### PR DESCRIPTION
**Pull request by dtzWill**
Naming "r2-cutter" since "cutter" already exists,
but AFAIK this naming is not used anywhere else currently.

###### Motivation for this change

Updated version of #32345 for 1.1 release (and new radare2 in #33194)

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by dtzWill** [comment]
(Soft ping :innocent:)

File: pkgs/development/tools/analysis/cutter/default.nix [hunk 1]
```diff
@@ -0,0 +1,49 @@
+{ stdenv, fetchFromGitHub, fetchpatch, qmake, pkgconfig, qtbase, qtsvg, radare2 }:
+
+
+let tag = "1.1"; in
+stdenv.mkDerivation rec {
+  name = "r2-cutter-${version}";
+  version = "${tag}.0";
```
Comment (orivej) [inline pkgs/development/tools/analysis/cutter/default.nix]: Why do you `.0` to the version number?

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -7707,6 +7707,7 @@ with pkgs;
     rubyBindings = config.radare.rubyBindings or false;
     luaBindings = config.radare.luaBindings or false;
   };
+  r2-cutter = qt5.callPackage ../development/tools/analysis/cutter { };
```
Comment (orivej) [inline pkgs/top-level/all-packages.nix]: Please use `libsForQt5`.

AUR names it `radare2-cutter` (https://aur.archlinux.org/packages/radare2-cutter/), would you like that name? I'd also rename the directory to `radare2-cutter` for consistency.

File: pkgs/development/tools/analysis/cutter/default.nix (hunk 1 above)
Comment (dtzWill) [reply pkgs/development/tools/analysis/cutter/default.nix]: Hmm, don't remember.  Something worked better with "full" version specification, maybe some tool I use.  If it seems odd I'll change it :).

File: pkgs/top-level/all-packages.nix (hunk 2 above)
Comment (dtzWill) [reply pkgs/top-level/all-packages.nix]: Aye aye.

**Comment by dtzWill** [comment]
Thanks for review @orivej! Updated to address your comments.

**Comment by joachifm** [comment]
@GrahamcOfBorg build radare2-cutter

**Comment by joachifm** [comment]
@dtzWill any idea about the aarch64 failure? If not, perhaps constrain the platform list to only x86 linux?

**Comment by dtzWill** [comment]
Looks to be another instance of https://github.com/nix-community/aarch64-build-box/issues/17, although this  is less clear than other cases.

**Comment by dtzWill** [comment]
(FWIW: 1.2 is out, but has a few bugs so waiting until 1.3 or whatever is next)

**Pull request merged**